### ISSUE TITLE
No way to navigate to hostnames

### ISSUE THREAD

**Issue by jellelicht** [atlas-engineer/nyxt#944]
**Describe the bug**
I would like nyxt to be able to navigate to `http://localhost:8383`, which is where I serve my blog locally before publising. Instead, it loads up a duckduckgo search for: `<INFO> [21:07:43] Loading "https://duckduckgo.com/?q=localhost:8383".
`
**Precise recipe to reproduce the issue**
**Information**
OS name+version: Guix System (master @ abafdb2aacefef0a2ecf46c8b06b52343727f34f)
Graphics card and driver: GM4500, default intel driver
Desktop environment / Window manager name+version: EXWM 0.24
How you installed Nyxt (Guix pack, package manager, build from source): guix package (2-pre-release-1)
Information from command copy-system-information:

```
Nyxt version: 2-pre-release-1 
Operating system kernel: Linux 4.19.142-gnu
Lisp implementation: SBCL 2.0.8
Features: (WEBKIT2 WEBKIT2-2.28 WEBKIT2-EMOJI WEBKIT2-MEDIA WEBKIT2-SANDBOXING
           GTK-3-22 GTK-3-20 GTK-3-18 GTK-3-16 GTK-3-14 GTK-3-12 GTK-3-10
           GTK-3-8 GTK-3-6 GTK-3-4 GTK GDK-3-22 GDK-3-20 GDK-3-18 GDK-3-16
           GDK-3-14 GDK-3-12 GDK-3-10 GDK-3-8 GDK-3-6 GDK-3-4 CAIRO-1-10
           CAIRO-1-12 GDK-PIXBUF GLIB-2-30 GLIB-2-32 GLIB-2-34 GLIB-2-36
           GLIB-2-38 GLIB-2-40 GLIB-2-42 GLIB-2-44 GLIB-2-46 GLIB-2-48
           GLIB-2-50 GLIB-2-52 GLIB-2-54 GLIB-2-56 GLIB-2-58 GLIB
           FSET-EXT-STRINGS SWANK PLUMP-UTF-32 GLOBAL-VARS DECLARE-TYPES
           PARENSCRIPT SBCL+SAFE-STANDARD-READTABLE NAMED-READTABLES
           OSICAT-FD-STREAMS LPARALLEL 21BIT-CHARS CL-FAD CHUNGA FLEXI-STREAMS
           CLOSER-MOP CL-PPCRE-UNICODE CL-UNICODE CL-PPCRE
           CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED CL-JSON-SINGLE-FLOAT-IS-SUBSUMED
           BORDEAUX-THREADS LPARALLEL.WITH-CLTL2 LPARALLEL.WITH-CAS
           LPARALLEL.WITH-STEALING-SCHEDULER SPLIT-SEQUENCE GRAY-STREAMS
           FLAT-NAMESPACE X86-64 UNIX CFFI FLAT-NAMESPACE SEQUENCE-EMPTYP
           FAST-IO-SV FAST-IO SBCL-USES-SB-ROTATE-BYTE CL-JSON-CLOS CL-JSON
           THREAD-SUPPORT ASDF3.3 ASDF3.2 ASDF3.1 ASDF3 ASDF2 ASDF OS-UNIX
           NON-BASE-CHARS-EXIST-P ASDF-UNICODE X86-64 GENCGC 64-BIT ANSI-CL
           COMMON-LISP ELF IEEE-FLOATING-POINT LINUX LITTLE-ENDIAN
           PACKAGE-LOCAL-NICKNAMES SB-CORE-COMPRESSION SB-LDB SB-PACKAGE-LOCKS
           SB-THREAD SB-UNICODE SBCL UNIX)
```

**Comment by Ambrevar** [comment]
Fixed in d865aa92e681436e8832052014bbd5faec63d666, thanks for reporting!

**Comment by jmercouris** [comment]
"Mitigate" was also referencing this on IRC. I thought we had fixed this some time ago? I remember developing with Caveman and Nyxt. Hm.

**Comment by jellelicht** [comment]
Note: I have no clue how resolving domain names is actually done in
practice.

Shouldn't _any_ valid hostname first be resolved?  e.g., on my machine,
the hostname is `braveheart`, so shouldn't this be fixed in a different
way? (That is at least how it works for me in Firefox)

**Comment by Ambrevar** [comment]
You are right, this would be the proper fix.  I don't know if WebKitGTK
can perform domain name resolution.

I'll reopen this issue.

**Comment by Ambrevar** [comment]
OK, found it, we just need to use `usocket:get-host-by-name`.
@jmercouris Sounds good to you?  If yes, I'll push a fix.

**Comment by jmercouris** [comment]
I'm very hesitant about including new dependencies. What does bringing in usocket require us to do?

**Comment by Ambrevar** [comment]
usocket is already in Nyxt.

**Comment by jmercouris** [comment]
??? I looked in the ASD, where is it?

**Comment by Ambrevar** [comment]
It's a dependency of dexador.
Nyxt drags way more than its direct dependencies! :p

**Comment by Ambrevar** [comment]
Fixed with fc25179113b6d419dbb2d35b1597b6365886855d.
